androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/SurveyDetailScreen.kt

```diff
@@ -62,10 +66,21 @@ fun SurveyDetailScreen(
                 onBackClick()
             }
         },
+        onQuestionAnswered = { questionUiModel ->
+            val index = questionsWithoutIntro?.indexOfFirst { it.id == questionUiModel.id }
```

Comment:
Ditto. I think that's why I am having a little bit of lagging while I scroll through the surveys.

Reply:
I think using a map would improve the performance. 🤔

Reply:
I can't reproduce the lag. Is it happening to all kinds of questions?

Reply:
@AVI5HEK It is more of smoothness than lagging. 
This is my IC. You can notice that the navigation from one page to another is smooth.
https://github.com/nimblehq/avishek-phong-kmm-ic/assets/32578035/34d02843-8d27-47da-b7ea-e5754dc3267f

This is KMM. You can notice that the navigation from one page to another is not as smooth as it should be.
https://github.com/nimblehq/avishek-phong-kmm-ic/assets/32578035/36c1b4b8-b444-434b-81f8-c92d651037ad

There are some API from compose like [derivedStateOf](https://medium.com/androiddevelopers/jetpack-compose-when-should-i-use-derivedstateof-63ce7954c11b) when you want to include filter logic inside Compose. But it is for only when you cannot avoid including data logic in the view.

Reply:
Here's the screenshot from `LayoutInspector`:
<img width="1197" alt="image" src="https://github.com/nimblehq/avishek-phong-kmm-ic/assets/8093908/a4f7ed56-49e2-478b-b422-3b84abbeccd9">
The button has been recomposed 28 times in only 3 pages.

If we set `userScrollEnabled = false` and change the pages by swiping manually, the transition doesn't seem too jittery. 🤔